src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEntry.Unix.cs

```diff
@@ -73,12 +75,11 @@ private ReadOnlySpan<char> FullPath
             {
                 if (_fullPath.Length == 0)
                 {
-                    ReadOnlySpan<char> directory = Directory;
-                    directory.CopyTo(_pathBuffer);
-                    _pathBuffer[directory.Length] = Path.DirectorySeparatorChar;
-                    ReadOnlySpan<char> fileName = FileName;
-                    fileName.CopyTo(_pathBuffer.Slice(directory.Length + 1));
-                    _fullPath = _pathBuffer.Slice(0, directory.Length + 1 + fileName.Length);
+                    Debug.Assert(Directory.Length + FileName.Length < _pathBuffer.Length,
+                        $"directory ({Directory.Length} chars) & name ({Directory.Length} chars) too long for buffer ({_pathBuffer.Length} chars)");
+                    Path.TryJoin(Directory, FileName, _pathBuffer, out int charsWritten);
+                    Debug.Assert(charsWritten > 0, "didn't write any chars to buffer");
```

Comment:
Should we also assert that TryJoin was successful?

Reply:
Too much bother. I didn't want to wrestle with release warnings or `#if DEBUG`. `0` should cover it.